src/calibrator.cc

```diff
@@ -123,33 +125,89 @@ bool Calibrator::finish(int width, int height)
         return false;
     }
 
+    // ========================================================================
+    // NEW RELIABILITY FIX:
+    // Re-sort click points to logical corners (UL, UR, LL, LR)
+    // before feeding them to the matrix solver. This fixes errors
+    // caused by screen rotation or mirroring.
+    // ========================================================================
+
+    struct Point { int x, y; };
+    std::vector<Point> original_clicks(NUM_POINTS);
+    for (int i = 0; i < NUM_POINTS; ++i) {
+        original_clicks[i] = {clicked_x[i], clicked_y[i]};
+    }
+
+    // Calculate the center point
+    float avg_x = (original_clicks[0].x + original_clicks[1].x + original_clicks[2].x + original_clicks[3].x) / 4.0f;
+    float avg_y = (original_clicks[0].y + original_clicks[1].y + original_clicks[2].y + original_clicks[3].y) / 4.0f;
+
+    std::vector<int> new_clicked_x(NUM_POINTS);
+    std::vector<int> new_clicked_y(NUM_POINTS);
+
+    for (const auto& click : original_clicks) {
+        if (click.x < avg_x && click.y < avg_y) {
+            // Top-Left (UL)
+            new_clicked_x[UL] = click.x;
+            new_clicked_y[UL] = click.y;
+        } else if (click.x > avg_x && click.y < avg_y) {
+            // Top-Right (UR)
+            new_clicked_x[UR] = click.x;
+            new_clicked_y[UR] = click.y;
+        } else if (click.x < avg_x && click.y > avg_y) {
+            // Bottom-Left (LL)
+            new_clicked_x[LL] = click.x;
+            new_clicked_y[LL] = click.y;
+        } else if (click.x > avg_x && click.y > avg_y) {
+            // Bottom-Right (LR)
+            new_clicked_x[LR] = click.x;
+            new_clicked_y[LR] = click.y;
+        }
+    }
+
+    // Atomically update the member variables
+    clicked_x = new_clicked_x;
+    clicked_y = new_clicked_y;
+
+    if (verbose) {
+        printf("DEBUG: Re-sorted click points (X, Y):\n");
+        printf("DEBUG: UL: (%d, %d)\n", clicked_x[UL], clicked_y[UL]);
+        printf("DEBUG: UR: (%d, %d)\n", clicked_x[UR], clicked_y[UR]);
+        printf("DEBUG: LL: (%d, %d)\n", clicked_x[LL], clicked_y[LL]);
+        printf("DEBUG: LR: (%d, %d)\n", clicked_x[LR], clicked_y[LR]);
+    }
+    // ========================================================================
+    // END OF FIX
+    // ========================================================================
+
+
     /*
      * Assuming that
      *
-     *  [a  b  c]     [tx_i]     [sx_i]
-     *  [d  e  f]  x  [ty_i]  =  [sy_i]
-     *  [0  0  1]     [  1 ]     [ 1  ]
+     * [a  b  c]     [tx_i]     [sx_i]
```

Comment:
Please preserve the spaces in the comments, otherwise these are not readable correctly.